splice_machine/src/main/java/com/splicemachine/derby/impl/sql/compile/NestedLoopJoinStrategy.java

```diff
@@ -61,6 +61,14 @@ public boolean feasible(Optimizable innerTable,
         if (outerCost != null && outerCost.getJoinType() == JoinNode.FULLOUTERJOIN)
             return false;
 
+        if (!isSingleTableScan(optimizer)) {
+            if (innerTable.isTriggerVTI())
+                return false;
+            Optimizable outerTable = optimizer.getOuterTable();
+            if (outerTable != null && outerTable.isTriggerVTI())
+                return false;
+        }
```

Comment:
From the PR commit message:

> and disallows use of nested loop join for statement triggers with a referencing clause on spark.

Do you want to strengthen the condition by adding `optimizer.isForSpark`?

In general, it would be nice to have a comment on why we don't want to use the NLJ in case of triggers (just to quickly read it without having to back to the PR itself :) ).

Reply:
We have to disable nested loop join even if not for spark.  I've added the following comment explaining why:

>     // Nested loop join on Spark does not work correctly when using a common
>     // Dataset to access the trigger REFERENCING NEW/OLD TABLE rows:
>     //         see useCommonDataSet in TriggerNewTransitionRows.
>     // The compilation of a trigger is saved as a stored prepared statement in
>     // the data dictionary, and reloaded/reused by each new triggering statement.
>     // Even if the trigger is compiled to run in OLTP mode, if the triggering
>     // statement runs in OLAP, the trigger must run in OLAP too.  Since we
>     // cannot tell from the SPSDescriptor whether the trigger was compiled
>     // to run on OLTP or OLAP, we would not be able to detect when an
>     // OLTP-compiled trigger which uses nested loop join would need to be
>     // recompiled as forced-OLAP, and avoid choosing nested loop join.
>     // Therefore we must always avoid nested loop join for statement triggers
>     // with a REFERENCING clause, even if compiled for OLTP execution.

Reply:
Great!